### PR TITLE
cmake: Headless version for macOS

### PR DESCRIPTION
**Pull request by jsakkine**
Add CMakeLists.txt that is able to build the headless version for macOS:

mkdir build
cmake -GXcode -Bbuild
cd build
xcodebuild
cd ..

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+)
+
+set(SURGE_COMMON_INCLUDES
+    src/common
+    src/common/dsp
+    src/common/gui
+    libs/xml
+    src/common/vt_dsp
+    src/common/thread
+    vst3sdk/vstgui4
```
Comment (baconpaul) [inline CMakeLists.txt]: This should definitely be vstgui.surge not vst3sdk/vstgui4 so we get our overridden vstgui

**Comment by jsakkine** [comment]
@baconpaul, can you give it a shot?  my cmake build file ever

**Comment by baconpaul** [comment]
Yup. Will give it a shot tonight

**Comment by baconpaul** [comment]
Oh did you see my comment about your using the wrong vstgui? You should use vstgui.surge not vst3sdk/vstgui4

Won’t matter here but will for proper plugins of course

**Comment by baconpaul** [comment]
Sorry @jsakkine I won't get to this tonight. Will look tomorrow in a break though.

**Comment by baconpaul** [comment]
Actually had 30 seconds so checked out 

```
paul:~/dev/music/surge$ cmake -GXcode -Bbuild
CMake Error: The source directory "" does not exist.
```

turns out my cmake is 3.1.something but the minimum version check didn't fire. I did a brew upgrade and it resolved.

**Comment by baconpaul** [comment]
Once I upgraded though I was able to build headless following your directions yes.

I'd be happy to merge this if we remove headless from premake and update the build-linux and build-osx to have commands to do this. Also change the vstgui location of course!

**Comment by jsakkine** [comment]
@baconpaul, OK whatya think?

**Comment by jsakkine** [comment]
I'll do Linux as a separate PR.

**Pull request merged**